### ISSUE TITLE
Read only error when running `mu4e-view-rendered-hook`

### ISSUE THREAD

**Issue by thierryvolpiatto** [djcb/mu#2500]
**Describe the bug**

The hook `mu4e-view-rendered-hook` runs outside the `inhibit-read-only` let body hence this error:

error in process filter: smiley-buffer: Buffer is read-only: #<buffer *mu4e-article*>

advising smiley-buffer which is added to mu4e-view-rendered-hook in my config fixes the bug.
 
**How to Reproduce**

Add smiley-buffer to mu4e-view-rendered-hook.

**Environment**

LinuxMint, emacs-28

**Checklist**
- [ X] you are running either an 1.8.x/1.10.x release or `master` (otherwise please upgrade)
- [ X] you can reproduce the problem without 3rd party extensions (including Doom/Evil, various extensions etc.)
- [ X] you have read all of the above

**Comment by djcb** [comment]
Yeah, making it read-write is probably the more useful behavior. I'll push a fix.
